test/test_torch.py

```diff
@@ -10779,9 +10779,32 @@ def test_tensor_factories_empty(self, device):
         self.assertEqual((0,), torch.bartlett_window(0, periodic=False, device=device).shape)
         self.assertEqual((0,), torch.hamming_window(0, device=device).shape)
         self.assertEqual((0,), torch.hann_window(0, device=device).shape)
+        self.assertEqual((0,), torch.vandermonde_matrix(torch.tensor([]), 0, device=device).shape)
         self.assertEqual((1, 1, 0), torch.tensor([[[]]], device=device).shape)
         self.assertEqual((1, 1, 0), torch.as_tensor([[[]]], device=device).shape)
 
+    @unittest.skipIf(not TEST_NUMPY, 'Numpy not found')
+    @dtypes(torch.long, torch.double)
+    def test_vandermonde_matrix(self, device, dtype):
+        x = [1, 2, 3, 5]
```

Comment:
Also for input values, maybe worth diverging integer and float inputs? To test numerical accuracy it might help to use some "weird" float values: (-pi, 0, 1/3, 1, pi, 3/7)?